addons/binding/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/internal/ssl/PinTrustManager.java

```diff
@@ -54,52 +52,33 @@ public void setCallback(PinnedCallback callback) {
     }
 
     @Override
-    public void checkClientTrusted(X509Certificate @Nullable [] chain, @Nullable String authType)
-            throws CertificateException {
-        throw new UnsupportedOperationException();
+    public X509Certificate[] getAcceptedIssuers() {
+        return new X509Certificate[0];
     }
 
-    protected byte[] getEncoded(PinType type, X509Certificate cert) throws CertificateEncodingException {
-        switch (type) {
-            case CERTIFICATE_TYPE:
-                return cert.getEncoded();
-            case PUBLIC_KEY_TYPE:
-                return cert.getPublicKey().getEncoded();
+    @Override
+    public void checkClientTrusted(X509Certificate @Nullable [] certs, @Nullable String authType)
+            throws CertificateException {
+        if ((certs == null) || (certs.length == 0)) {
+            throw new IllegalArgumentException();
         }
-        throw new CertificateEncodingException("Type unknown");
-    }
 
-    /**
-     * A signature name depends on the security provider but usually follows
-     * https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#Signature.
-     * E.g.: "SHA256withRSA". We need "SHA" and "256" to initialize a {@link PinMessageDigest}.
-     */
-    PinMessageDigest getMessageDigestForSigAlg(String sigAlg) throws CertificateException {
-        final Matcher matcher = Pattern.compile("(\\D*)(\\d+)").matcher(sigAlg);
-        matcher.find();
-        final String sigAlgName = matcher.group(1);
-        final String sigAlgBits = matcher.group(2);
-        try {
-            return new PinMessageDigest(sigAlgName + "-" + sigAlgBits);
-        } catch (NoSuchAlgorithmException e) {
-            throw new CertificateException(e);
-        }
+        throw new UnsupportedOperationException();
     }
 
     @Override
-    public void checkServerTrusted(X509Certificate @Nullable [] chainN, @Nullable String authType)
+    public void checkServerTrusted(X509Certificate @Nullable [] certs, @Nullable String authType)
             throws CertificateException {
-        X509Certificate[] chain = chainN;
-        if (chain == null) {
-            return;
+        if ((certs == null) || (certs.length == 0)) {
```

Comment:
There was a reason for this variable. It must be null checked so that our static null checker knows that it is safe to be used in the methods below. In the signature you'll see that `certs` can be null.

Reply:
As far i can see, certs is still check against null. Miss i something?
![grafik](https://user-images.githubusercontent.com/19878123/52504648-60a04000-2be9-11e9-9682-19e9e3c8abc7.png)

Reply:
Just judging by logic, it should work in this short variant. There was a time when the static null checker required a declared non @Nullable variable. Have you also build with maven (/checked the travis output)? The output contains the static code checker results. I'd say we risk it and take this new code.

Reply:
As far i've seen, there was no finding about this. I'll cross check again.

Reply:
Jenkins is happy: https://ci.openhab.org/job/PR-openHAB2-Addons/12398/Static_20Code_20Analysis_20Report/